usr/share/rear/verify/BAREOS/default/050_check_requirements.sh

```diff
@@ -1,34 +1,45 @@
 #
-# Check that bareos is installed and configuration files exist
+# Check that bareos is configuration exist
 
-# First determine whether we need to restore using bconsole or bextract.
+LogPrint "Bareos: checking requirements ..."
 
-if [ "$BEXTRACT_DEVICE" -o "$BEXTRACT_VOLUME" ]; then
+# First determine whether we need to restore using bconsole or bextract.
 
-   ### Bareos support using bextract
-   if [ -z "$BEXTRACT_VOLUME" ]; then
-      BEXTRACT_VOLUME=*
-   fi
+if [ "$BAREOS_RESTORE_MODE" = "bextract" ]; then
 
-   [ -x /usr/sbin/bextract ]
-   StopIfError "Bareos executable (bextract) missing or not executable"
+    ### Bareos support using bextract
+    if [ -z "$BEXTRACT_VOLUME" ]; then
+        BEXTRACT_VOLUME="*"
+    fi
 
-   bareos-sd -t 
-   StopIfError "Bareos-sd configuration invalid"
+    if ! bareos-sd -t; then
+        Error "Bareos-sd configuration invalid"
+    fi
 
 else
 
-   ### Bareos support using bconsole
-   [ -x /usr/sbin/bareos-fd ]
-   StopIfError "Bareos executable (bareos-fd) missing or not executable"
+    ### Bareos support using bconsole
+
+    if ! bareos-fd -t; then
+        Error "bareos-fd: configuration invalid"
+    fi
 
-   bareos-fd -t
-   StopIfError "Bareos-fd configuration invalid"
+    if ! systemctl --quiet is-active bareos-fd.service; then
+        Log "$(systemctl status bareos-fd.service)"
+        Error "bareos-fd.service is not running"
+    fi
 
-   [ -x /usr/sbin/bconsole ]
-   StopIfError "Bareos executable (bconsole) missing or not executable"
+    while ! systemctl is-active bareos-fd.service; do
```

Comment:
If the `Error` in line 29 hits, how will we get here?

Maybe instead of checking if it is running just try to start it and rely on systemd to do the right thing?